devdoc/rc_string_requirements.md

```diff
@@ -72,6 +72,33 @@ MOCKABLE_FUNCTION(, THANDLE(RC_STRING), rc_string_create, const char*, string);
 
 **SRS_RC_STRING_01_006: [** If any error occurs, `rc_string_create` shall fail and return `NULL`. **]**
 
+## rc_string_create_with_format
+
+```c
+THANDLE(RC_STRING) rc_string_create_with_format(const char* format, ...);
+```
+
+`rc_string_create_with_format` creates a new ref counted string by using the format convention as in sprintf.
+
+If `format` is `NULL`, `rc_string_create_with_format` shall fail and return `NULL`. 
+
+Otherwise, `rc_string_create_with_format` shall determine the total number of characters written using the variadic arguments. 
+
+`rc_string_create_with_format` shall allocate memory for the formatted string including the `NULL` terminator.
+
+`rc_string_create_with_format` shall copy the formatted string result including the `NULL` terminator into `string`. 
+
+`rc_string_create_with_format` shall allocate memory for  the `THANDLE(RC_STRING)`, ensuring all the bytes in `string` can be copied (including the zero terminator).
```

Comment:
So, why do we need 2 allocations? I would like this to be done with 1 allocation. I think we know how to do that all the time:
- get the length of the needed string
- allocate the memory for the THANDLE(RC_STRING) AND the string that it needs to contain
- fill in the alias pointer in the THANDLE(RC_STRING)
- fill in the bytes of the string by using vsnprintf

Reply:
I shall wait for specs to be changed accordingly.